Mathlib/Probability/Moments/IntegrableExpMul.lean

```diff
@@ -0,0 +1,331 @@
+    · simp only [Set.mem_setOf_eq, h_nonneg, Set.indicator_of_mem]
+      calc rexp (t * X ω) ≤ 1 + rexp (t * X ω) := (le_add_iff_nonneg_left _).mpr zero_le_one
+      _ ≤ 1 + exp (u * X ω) := by gcongr
+    · simp only [Set.mem_setOf_eq, not_le.mpr h_neg, not_false_eq_true, Set.indicator_of_not_mem,
+        add_zero, exp_le_one_iff]
+      exact mul_nonpos_of_nonneg_of_nonpos h_pos.le h_neg.le
+
+/-- If `ω ↦ exp (u * X ω)` is integrable at `u ≥ 0`, then it is integrable on `[0, u]`. -/
+lemma integrable_exp_mul_of_nonneg_of_le [IsFiniteMeasure μ]
+    (hu : Integrable (fun ω ↦ exp (u * X ω)) μ) (h_nonneg : 0 ≤ t) (htu : t ≤ u) :
+    Integrable (fun ω ↦ exp (t * X ω)) μ := by
+  by_cases ht : t = 0
+  · simp [ht]
+  have hX : AEMeasurable X μ := by
+    refine aemeasurable_of_aemeasurable_exp_mul ?_ hu.aemeasurable
+    exact ((lt_of_le_of_ne' h_nonneg ht).trans_le htu).ne'
+  have h_eq t : (fun ω ↦ exp (t * X ω)) =ᵐ[μ] fun ω ↦ exp (t * hX.mk X ω) := by
+    filter_upwards [hX.ae_eq_mk] with ω hω using by rw [hω]
+  rw [integrable_congr (h_eq t)]
+  rw [integrable_congr (h_eq u)] at hu
+  exact integrable_exp_mul_of_le_of_measurable hX.measurable_mk hu h_nonneg htu
+
+/-- If `ω ↦ exp (u * X ω)` is integrable at `u ≤ 0`, then it is integrable on `[u, 0]`. -/
+lemma integrable_exp_mul_of_nonpos_of_ge [IsFiniteMeasure μ]
+    (hu : Integrable (fun ω ↦ exp (u * X ω)) μ) (h_nonpos : t ≤ 0) (htu : u ≤ t) :
+    Integrable (fun ω ↦ exp (t * X ω)) μ := by
+  suffices Integrable (fun ω ↦ exp (- t * (-X) ω)) μ by simpa using this
+  exact integrable_exp_mul_of_nonneg_of_le (u := -u) (t := -t)
+    (by simpa using hu) (by simp [h_nonpos]) (by simp [htu])
+
+/-- If `ω ↦ exp (u * X ω)` is integrable at `u` and `-u`, then it is integrable on `[-u, u]`. -/
+lemma integrable_exp_mul_of_abs_le [IsFiniteMeasure μ]
+    (hu_int_pos : Integrable (fun ω ↦ exp (u * X ω)) μ)
+    (hu_int_neg : Integrable (fun ω ↦ exp (- u * X ω)) μ)
+    (htu : |t| ≤ |u|) :
+    Integrable (fun ω ↦ exp (t * X ω)) μ := by
+  rcases le_total 0 t with ht | ht
+  · rw [abs_of_nonneg ht] at htu
+    refine integrable_exp_mul_of_nonneg_of_le ?_ ht htu
+    rcases le_total 0 u with hu | hu
+    · rwa [abs_of_nonneg hu]
+    · rwa [abs_of_nonpos hu]
+  · rw [abs_of_nonpos ht, neg_le] at htu
+    refine integrable_exp_mul_of_nonpos_of_ge ?_ ht htu
+    rcases le_total 0 u with hu | hu
+    · rwa [abs_of_nonneg hu]
+    · rwa [abs_of_nonpos hu, neg_neg]
+
+lemma integrable_exp_mul_of_le_of_le [IsFiniteMeasure μ] {a b : ℝ}
+    (ha : Integrable (fun ω ↦ exp (a * X ω)) μ) (hb : Integrable (fun ω ↦ exp (b * X ω)) μ)
+    (hat : a ≤ t) (htb : t ≤ b) :
+    Integrable (fun ω ↦ exp (t * X ω)) μ := by
+  rcases le_total 0 t with ht | ht
+  · exact integrable_exp_mul_of_nonneg_of_le hb ht htb
+  · exact integrable_exp_mul_of_nonpos_of_ge ha ht hat
+
+end Interval
+
+section FiniteMoments
+
+lemma integrable_exp_mul_abs_add (ht_int_pos : Integrable (fun ω ↦ rexp ((v + t) * X ω)) μ)
```

Comment:
The exponential is called `exp` before, is there a reason you switch to `rexp` here?

Reply:
I guess I copied something from the infoview, which used rexp. I'll switch it back to exp.